scripts/ci/build_emscripten.sh

```diff
@@ -34,55 +34,101 @@
-# default in clang-16 which is what the emscripten docker image uses.
-# Additionally, disable the warning for unknown warnings here, as this script is
-# also used with earlier clang versions.
-# TODO: This can be removed if and when all usages of `move()` in our codebase use the `std::` qualifier.
-CMAKE_CXX_FLAGS="-Wno-unqualified-std-cast-call"
-
-
-mkdir -p "$BUILD_DIR"
-cd "$BUILD_DIR"
-emcmake cmake \
-	-DCMAKE_BUILD_TYPE=Release \
-	-DBoost_USE_STATIC_LIBS=1 \
-	-DBoost_USE_STATIC_RUNTIME=1 \
-	-DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS}" \
-	-DTESTS=0 \
-  ..
-make soljson
-
-cd ..
-mkdir -p upload
-scripts/ci/pack_soljson.sh "$BUILD_DIR/libsolc/soljson.js" "$BUILD_DIR/libsolc/soljson.wasm" upload/soljson.js
-cp upload/soljson.js ./
-
-OUTPUT_SIZE=$(ls -la soljson.js)
-
-echo "Emscripten output size: $OUTPUT_SIZE"
+function build() {
+    local build_dir=$1
+    local pre_release=$2
+
+    cd $WORKSPACE
+
+    # shellcheck disable=SC2166
+    if [[ "$CIRCLE_BRANCH" = release || -n "$CIRCLE_TAG" || -n "$FORCE_RELEASE" || "$(git tag --points-at HEAD 2>/dev/null)" == v* ]]
+    then
+        echo -n >prerelease.txt
+    else
+        # Use the given prerelease string if provided
+        if [[ "$pre_release" =~ ^[a-zA-Z]+\.[0-9]{4}\.[0-9]{1,2}\.[0-9]{1,2}$ ]]; then
+            echo "$pre_release" >prerelease.txt
+        else
+            # Use last commit date rather than build date to avoid ending up with builds for
+            # different platforms having different version strings (and therefore producing different bytecode)
+            # if the CI is triggered just before midnight.
+            TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
+        fi
+    fi
+    if [ -n "$CIRCLE_SHA1" ]
+    then
+        echo -n "$CIRCLE_SHA1" >commit_hash.txt
+    fi
+
+    # Disable warnings for unqualified `move()` calls, introduced and enabled by
+    # default in clang-16 which is what the emscripten docker image uses.
+    # Additionally, disable the warning for unknown warnings here, as this script is
+    # also used with earlier clang versions.
+    # TODO: This can be removed if and when all usages of `move()` in our codebase use the `std::` qualifier.
+    CMAKE_CXX_FLAGS="-Wno-unqualified-std-cast-call"
+
+
+    mkdir -p "$build_dir"
+    cd "$build_dir"
+    emcmake cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBoost_USE_STATIC_LIBS=1 \
+        -DBoost_USE_STATIC_RUNTIME=1 \
+        -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS}" \
+        -DTESTS=0 \
+    ..
+    make soljson
+
+    cd ..
+    mkdir -p upload
+    scripts/ci/pack_soljson.sh "$build_dir/libsolc/soljson.js" "$build_dir/libsolc/soljson.wasm" upload/soljson.js
+    cp upload/soljson.js ./
+
+    OUTPUT_SIZE=$(ls -la soljson.js)
+
+    echo "Emscripten output size: $OUTPUT_SIZE"
+}
+
+function show_help() {
+cat << EOF
+Usage: ${0##*/} [-h|--help] [--build-dir DIR] [--prerelease release_type.YYYY.MM.DD]
```

Comment:
The builds we use this for are actually never releases.
```suggestion
Usage: ${0##*/} [-h|--help] [--build-dir DIR] [--prerelease prerelease_source.YYYY.MM.DD]
```